File: mergeTwoSortedLists/sol.c
struct ListNode* mergeTwoLists(struct ListNode* l1, struct ListNode* l2){
  if (l1 == NULL) return l2;

  struct ListNode* head = l1;
  struct ListNode* prev = NULL;

  while (l2 != NULL)
    if (l1->val <= l2->val) {
      prev = l1;
      if (l1->next != NULL)
        l1 = l1->next;
      else {
        l1->next = l2;
        break;
      }
    }
    else {
      if (prev == NULL) {
        head = l2;
        l2 = l2->next;
        head->next = l1;
        prev = head;
      } else {
        prev->next = l2;
        l2 = l2->next;
        prev = prev->next;
        prev->next = l1;
      } 
    }
  return head;
}

File: problems/Middle_of_the_Linked_List/sol.go
package main

type ListNode struct {
  Val int
  Next *ListNode
}

func middleNode(head *ListNode) *ListNode {
  // Find the length of the list.
  var l int
  lp := head
  for l = 0; lp.Next != nil; l++ {
    lp = lp.Next
  }
  // Now go to the middle.
  lp = head
  for i := 0; i <= l/2 ; i++ {
    lp = lp.Next
  }
  return lp
}
